### PR TITLE
Print something readable for problems with more than two parameters

### PR DESCRIPTION
**Pull request by johnmyleswhite**
tl;dr: This makes printing not suck when you have a large parameter space.

### CODE REVIEW

**Comment by mlubin** [comment]
Looks like a good improvement. Could you add some tests for this? I don't think the printing code has any coverage.

**Comment by johnmyleswhite** [comment]
Sure, I can do that. I'm generally pretty cautious about adding rich tests for show methods because they're so brittle and tedious. I'll probably just check that there's a line for each expected entry in the method.

**Comment by mlubin** [comment]
Works for me, thanks!

**Comment by pkofod** [comment]
Any chance we can address minimum vs minimizer in the output as per #76 ?

**Comment by johnmyleswhite** [comment]
Sure. I'd also be very open to a PR that adds a real and documented API. I keep dreaming making a better Optim someday with a better API, but we could make substantial progress if someone puts in a few days of work.

**Comment by pkofod** [comment]
Yeah, I agree. A lot of great work has already been done, but Optim is still very unpolished. At some point I'd be more than willing to donate a some days worth of work - but as for everyone else time is not in surplus here at the moment. We have no active issue/wishlish to discuss the overall future of Optim, right?

**Comment by johnmyleswhite** [comment]
Not to my knowledge. But Optim is actually remarkably reliable given how easily it could have bugs and how many people seem to use it. I'm always surprised that Optim works as well as it does.

My personal priorities would be:

* A real API based on functions, not accessing fields of returned objects
* Real documentation -- both on the web and in the REPL
* Better correctness tests
* Better perf tests

The really ambitious project would be a redesign of the API to handle some kinds of constrained optimization problems. I would also like to rethink the assumptions we make about the functions we take in as arguments. But I think we should avoid going down that path until other stuff has been done.

**Comment by pkofod** [comment]
I agree that it is quite reliable, but re:unpolished I was thinking about more superficial things. As you mentioned, stuff like functions to extract information, documentation, etc. I agree with your list (order as well), although perf tests are so difficult to "do right". We should be able to show some minimal results on performance though.

I'll see if I can throw some time at this when I get home from Canada.

**Comment by johnmyleswhite** [comment]
Added a test for the new `show` method. Once tests pass, I'll merge.

File: test/types.jl [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module TestTypes
+    using Base.Test
+    using Optim
```
Comment (tkelman) [inline test/types.jl]: using `Compat`. or drop 0.3 if you're going to do that anyway

**Comment by johnmyleswhite** [comment]
Just threw that in after seeing the failures.

**Pull request merged**